pinot-core/src/main/java/com/linkedin/pinot/core/operator/aggregation/ResultHolderFactory.java

```diff
@@ -30,7 +30,20 @@
   private ResultHolderFactory() {
   }
 
-  public static final int MAX_INITIAL_RESULT_HOLDER_CAPACITY = 10000;
+  public static final int MAX_INITIAL_RESULT_HOLDER_CAPACITY = 10_000;
+  public static final int MAX_NUM_GROUPS_ALLOWED;
+
+  // Read the max number of groups allowed from unix env variable.
+  static {
+    int numGroups;
+    try {
+      numGroups =
+          Integer.parseInt(System.getProperty("pinot.max.aggr.groups"));
```

Comment:
This should be a config property, not an environment property. All other Pinot settings come from config files.